lib/index.js

```diff
@@ -22,36 +25,49 @@ const createRule = (name, browserstring, description, { ts = null } = {}) => {
     ),
     create(context) {
       let browsers = browserslist(browserstring);
-      const config = findConfig(path.dirname(context.getFilename())) || {
+      const config = findConfig(path.dirname(
+        context.filename ?? context.getFilename()
+      )) || {
         defaults: "defaults",
       };
       const desiredBrowsers = browserslist(config.defaults);
       const badBrowsers = desiredBrowsers
-        .filter((browser) => browsers.indexOf(browser) !== -1)
+        .filter((browser) => browsers.includes(browser))
         .join(", ");
-      if (badBrowsers) {
+      if (globalThis.ESLINT_TESTING || badBrowsers) {
         const create = typeof rule === "function" ? rule : rule.create;
-        return create(context, badBrowsers);
+        return create(context, globalThis.ESLINT_TESTING ? undefined : badBrowsers);
       }
       return {};
     },
   };
 
   const configName = `typescript-${ts || "base"}`;
+  const flatConfigName = `flat/${configName}`;
```

Comment:
I haven't read much on flat configs but does it need to expose a whole new namespace? Can you link me to some literature on this change please.

Reply:
It needs a new object format, so if you want to keep supporting ESLint prior to 9, and you're doing more than exporting rules such as exporting plugins, one has to export in such a manner. 

A page detailing the fact that the object format changes is at https://eslint.org/docs/latest/extend/plugin-migration-flat-config#migrating-configs-for-flat-config

Reply:
It seems like we want to expose this in the documentation then, no? Because those using flat configs would need to use `flat/recommended` instead of `recommended`?

Reply:
Sure. I didn't enumerate all the TypeScript configs as you hadn't done so, but I added a section on flat config usage and showed how to use with the TypeScript configs in general.

I also had neglected to auto-add plugins for the TypeScript configs, now added.